flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGateTest.java

```diff
@@ -226,7 +225,8 @@ public void testConcurrentReadStateAndProcessAndClose() throws Exception {
 						if (!getNextBufferAndVerify(inputGate, states)) {
 							Thread.sleep(1);
 						}
-					} catch (CancelTaskException expected) {
+					} catch (Throwable t) {
+						assertTrue("The gate is expected closed, but the actual exception is " + t.getMessage(), inputGate.getCloseFuture().isDone());
```

Comment:
The error message I think is incorrect, as I guess the point is, if the close future is NOT done, no exception should occur? 
```
if (!inputGate.getCloseFuture().isDone()) {
  throw new AssertionError("Exceptions are expected here only if the gate had been closed", t);
}
```

Reply:
Yes, the exception should only happen in the case of closed gate. Your error message seems better.